### PR TITLE
Editor / CPT: Migrate notices to component using type object label

### LINKED ISSUE #3695: CPT / Editor: Ensure editor uses post type labels when applicable
Post types can define labels to be shown in the interface. In the current editor, many labels are hard-coded to either the post or page context. We should update these labels to pull from the post type API response when possible.

#### Return to list view

image

From: `postType.labels.menu_name`

#### New post header label

image

From: `postType.labels.new_item`

#### Status label loading text

image

https://github.com/Automattic/wp-calypso/blob/6dde06c/client/post-editor/status-label.jsx#L62-L68

This should be updated to either `Loading…` or `Loading %(postType)s…`

From: `postType.labels.singular_name` (lowercase)

### PR DESCRIPTION
**Pull request by aduth**
Related: #5987
Closes #3695

This pull request seeks to refactor editor notices, moving the render logic to a separate component `<EditorNotice />`. In doing so, the notices are now aware of the the labels associated with a post type. Specifically, we now use the `view_item` label of the post type when displaying a "View Post" action after publishing a post.

Before|After
---|---
Before|After

__Implementation notes:__

I plan to include tests for this new component, but had encountered a few issues in the process of doing so. I'll plan to push up a commit including tests before merging these changes.

__Testing instructions:__

Verify that there are no regressions for all existing notices, including success (published, updated) and failures (no content, unknown failure).

1. Navigate to the [post editor](http://calypso.localhost:3000/post) (preferably the [portfolio project editor](http://calypso.localhost:3000/edit/jetpack-portfolio) if enabled to demonstrate view label)
2. Add a title and/or content
3. Publish the post
4. Note that the notice shown appears exactly as it does in master for posts, or using the `view_item` label if applicable for a custom post type

Test live: https://calypso.live/?branch=add/editor-notices

### CODE REVIEW

**Comment by aduth** [comment]
The rebased 55b453a includes tests for `<EditorNotice />`

In 3fe64f3 , I installed a new Chai test helper [`chai-enzyme`](https://github.com/producthunt/chai-enzyme) for additional Enzyme-related assertions. This required bumping related test dependencies to meet peer dependency requirements. Chai 3.0.0 caused an existing test to start failing which appeared to have been an invalid test (cc @ebinnion to verify, from 82921ef1ca59287d3da1ec86bb0fb2b08fc04e8f).

**Comment by ebinnion** [comment]
Based on the action [here](https://github.com/Automattic/wp-calypso/blob/master/client/state/jetpack-connect/actions.js#L341), it looks like your change is correct. I'm not sure why the test wasn't failing before. 😱

**Comment by aduth** [comment]
@ebinnion I assume that's what the breaking change in 3.0.0 was trying to fix, since the previous behavior was a bit confusing:

>`.has.property('foo', undefined)` will now specifically test to make sure that the value `'foo'` is actually undefined - before it simply tested that the key existed (but could have a value).

https://github.com/chaijs/chai/releases/tag/3.0.0

File: client/state/jetpack-connect/test/reducer.js [hunk 1]
```diff
@@ -282,7 +282,7 @@ describe( 'reducer', () => {
 
 			const state = jetpackSSO( undefined, action );
 
-			expect( state ).to.have.property( 'ssoUrl', action.sso_url );
+			expect( state ).to.have.property( 'ssoUrl', action.ssoUrl );
```
Comment (timmyc) [inline client/state/jetpack-connect/test/reducer.js]: sneaky

**Comment by timmyc** [comment]
This is testing out well for me.  🚢 all the things.

**Comment by alisterscott** [comment]
Hi, I just wanted to confirm that it's correct that the class of this notice has now changed.
We used to refer to this in our e2e tests as `.post-editor-notice.is-success` and now it looks like it's just `.notice.is-success`, Is this expected?

publish selector

**Comment by aduth** [comment]
@alisterscott That would be expected, yes. Specifically, it would be `.editor-notice .notice.is-success`

https://github.com/Automattic/wp-calypso/pull/6794/files#diff-87872f4c82a803e639bc81200e352d6eR160

**Pull request merged**